JotunnLib/Entities/CustomRoom.cs

```diff
@@ -0,0 +1,172 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Jotunn.Configs;
+using Jotunn.Managers;
+using SoftReferenceableAssets;
+using UnityEngine;
+
+namespace Jotunn.Entities
+{
+    /// <summary>
+    ///     Main interface for adding custom dungeon rooms to the game.<br />
+    ///     All custom rooms have to be wrapped inside this class to add it to Jötunns <see cref="DungeonManager"/>.
+    /// </summary>
+    public class CustomRoom : CustomEntity
+    {
+        /// <summary>
+        ///     The prefab for this custom room.
+        /// </summary>
+        public GameObject Prefab { get; }
+
+        /// <summary>
+        ///     The <see cref="global::Room"/> component for this custom room as a shortcut.
+        /// </summary>
+        public Room Room { get; }
+
+        /// <summary>
+        ///     The name of this custom room as a shortcut.
+        /// </summary>
+        public string Name { get; }
+
+        /// <summary>
+        ///     Indicator if references from <see cref="Entities.Mock{T}"/>s will be replaced at runtime.
+        /// </summary>
+        public bool FixReference { get; set; }
+
+        /// <summary>
+        ///     Theme name of this room.
+        /// </summary>
+        public string ThemeName { get; private set; }
+        
+        /// <summary>
+        ///     <see cref="Room.Theme"/> of this room.
+        /// </summary>
+        public Room.Theme Theme { get; private set; }
+
+        /// <summary>
+        ///     Associated <see cref="DungeonDB.RoomData"/> holding data used during generation.
+        /// </summary>
+        public DungeonDB.RoomData RoomData { get; private set; }
+
+
+        /// <summary>
+        ///     Custom room from a prefab loaded from an <see cref="AssetBundle"/> with a <see cref="global::Room"/> made from a <see cref="RoomConfig"/>.<br />
+        ///     Can fix references for <see cref="Entities.Mock{T}"/>s.
+        /// </summary>
+        /// <param name="assetBundle">A preloaded <see cref="AssetBundle"/></param>
+        /// <param name="assetName">Name of the prefab in the bundle.</param>
+        /// <param name="roomConfig">The config for this custom room.</param>
+        public CustomRoom(AssetBundle assetBundle, string assetName, RoomConfig roomConfig)
+            : this(assetBundle, assetName, false, roomConfig) { }
```

Comment:
Constructors without a `fixReference` are considered obsolete and should not be provided in new interfaces for consistency

Reply:
Addressed in: https://github.com/Valheim-Modding/Jotunn/pull/434/commits/5fb11c61edd45d61c090db6a73e9967e53e0c3f0